From a5642827bac9499a988157d6e1bb539adf5911e5 Mon Sep 17 00:00:00 2001
From: Josh Wilsdon <jwilsdon@joyent.com>
Date: Mon, 8 Apr 2019 20:16:32 -0700
Subject: [PATCH] TRITON-1367 Update VMAPI to node v6.x (again) Reviewed by:
 Trent Mick <trentm@gmail.com> Approved by: Trent Mick <trentm@gmail.com>

---
 Makefile     |  2 +-
 lib/vmapi.js | 18 ++++++++----------
 2 files changed, 9 insertions(+), 11 deletions(-)

diff --git a/Makefile b/Makefile
index d4a67b1..3f85ad5 100644
--- a/Makefile
+++ b/Makefile
@@ -24,7 +24,7 @@
 
 # The prebuilt sdcnode version we want. See
 # "tools/mk/Makefile.node_prebuilt.targ" for details.
-NODE_PREBUILT_VERSION=v4.9.0
+NODE_PREBUILT_VERSION=v6.17.0
 ifeq ($(shell uname -s),SunOS)
 	NODE_PREBUILT_TAG=zone
 	# Allow building on other than sdc-minimal-multiarch-lts@15.4.1
diff --git a/lib/vmapi.js b/lib/vmapi.js
index 8ff57fc..6db1a56 100644
--- a/lib/vmapi.js
+++ b/lib/vmapi.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright 2018, Joyent, Inc.
+ * Copyright 2019, Joyent, Inc.
  */
 
 /*
@@ -218,10 +218,10 @@ VmapiApp.prototype._initApis = function _initApis(options) {
             // interesting and is usually big.
             //
             // 2. A HEAD request: even if a body is set in this case, it's only
-            // so that Content-Length and Content-MD5 response headers are
-            // properly set in the custom JSON restify formatter according to
-            // RFC 2616. Having it in the audit log is not relevant since it's
-            // actually not sent to the client sending the request.
+            // so that the Content-Length header is properly set in the custom
+            // JSON restify formatter according to RFC 2616. Having it in the
+            // audit log is not relevant since it's actually not sent to the
+            // client sending the request.
             //
             // 3. A PutVms request: the response's body would be equivalent to
             // the request's data and is usually big.
@@ -442,17 +442,15 @@ function formatJSON(req, res, body, callback) {
     }
 
     var data = JSON.stringify(body);
-    var md5 = crypto.createHash('md5').update(data).digest('base64');
 
     res.setHeader('Content-Length', Buffer.byteLength(data));
-    res.setHeader('Content-MD5', md5);
     res.setHeader('Content-Type', 'application/json');
 
     if (req.method === 'HEAD') {
         // In case of a successful response to a HEAD request, the formatter is
-        // used to properly set the Content-Length and Content-MD5 headers, but
-        // no data should actually be sent as part of the response's body. This
-        // is all according to RFC 2616.
+        // used to properly set the Content-Length header, but no data should
+        // actually be sent as part of the response's body. This is all
+        // according to RFC 2616.
         formattedJson = '';
     } else {
         formattedJson = data;
-- 
2.21.0

